### PR TITLE
Fix #2080: Remove injected TestCoroutineDispatchers if not in use

### LINKED ISSUE #2080: Remove injected TestCoroutineDispatchers if not in use
**Describe the bug**
There are some test files where we are injecting `TestCoroutineDispatchers` but not using it. 

**Expected behavior**
Remove injected `TestCoroutineDispatchers` if not in use. 

**How to Fix this Issue**
1. Check the injected `TestCoroutineDispatchers` is used in test cases or not. If it is not in use you will find only we place where it is defined as a) variable init, b) inside `setUp` and c) inside `tearDown`. 
2. Remove these 3 initialisation and function calling of `TestCoroutineDispatchers`.
3. Run the test in Espresso and Robolectric to check all test are passing. Wiki reference - https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing
4. If the test cases are failing without `TestCoroutineDispatchers`, we don't need to change that particular file. This will be a rare edge case. 
4. When you send a PR for this issue, also attach both the screenshot showing the tests are passing in both Espresso and Robolectric. 
5. If there are tests `@Ignore` leave them unchanged.

## List of FIles:
- [ ] `FAQSingleActivityTest` 
- [ ] `FAQListFragmentTest` 
- [ ] `HelpFragmentTest` 
- [ ] `MyDownloadsFragmentTest` 
- [ ] `HtmlParserTest`

### PR DESCRIPTION
**Pull request by ShridharGoel**
## Explanation
Fixes #2080 

Removed injected `TestCoroutineDispatchers` from the following test since it was not being used:
1. FAQListFragmentTest
2. FAQSingleActivityTest
3. HelpFragmentTest
4. HtmlParserTest
5. MyDownloadsFragmentTest

## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### Roboelectric:

FAQListFragmentTest
FAQSingleActivityTest
HelpFragmentTest
HtmlParserTest
MyDownloadsFragmentTest

### Espresso:

FAQListFragmentTest
FAQSingleActivityTest
HelpFragmentTest
HtmlParserTest
MyDownloadsFragmentTest

### CODE REVIEW

**Comment by anandwana001** [comment]
@ShridharGoel You had to complete the onboarding instructions first, else PR will be closed by bot.

**Comment by ShridharGoel** [comment]
@anandwana001 I have done it now. Can you reopen the PR? I don't have the option to reopen it.

**Comment by anandwana001** [comment]
@ShridharGoel Please fix the ktlint failure. Check wiki guide here - https://github.com/oppia/oppia-android/wiki/Ktlint-Guide

**Comment by ShridharGoel** [comment]
@anandwana001 I have removed the unused imports. That has fixed it.

**Comment by anandwana001** [comment]
> @anandwana001 I have removed the unused imports. That has fixed it.

Could you also put Espresso tests result of these classes!!

File: app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsFragmentTest.kt [hunk 1]
```diff
@@ -63,27 +62,20 @@ import org.oppia.android.util.parser.HtmlParserEntityTypeModule
...
 
 /** Tests for [MyDownloadsFragment]. */
 @RunWith(AndroidJUnit4::class)
 @LooperMode(LooperMode.Mode.PAUSED)
 @Config(application = MyDownloadsFragmentTest.TestApplication::class, qualifiers = "port-xxhdpi")
 class MyDownloadsFragmentTest {
-  @Inject
-  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
-
   @Before
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsFragmentTest.kt:72]: leave a line above.

File: app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsFragmentTest.kt [hunk 2]
```diff
@@ -63,27 +62,20 @@ import org.oppia.android.util.parser.HtmlParserEntityTypeModule
...
   fun setUp() {
     setUpTestApplicationComponent()
-    testCoroutineDispatchers.registerIdlingResource()
   }
 
   @After
-  fun tearDown() {
-    testCoroutineDispatchers.unregisterIdlingResource()
-  }
+  fun tearDown() {}
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsFragmentTest.kt]: no need for this function now.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @ShridharGoel 
nit changes suggested

File: app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsFragmentTest.kt (hunk 1 above)
Comment (ShridharGoel) [reply app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsFragmentTest.kt:72]: Updated.

File: app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsFragmentTest.kt (hunk 2 above)
Comment (ShridharGoel) [reply app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsFragmentTest.kt]: Updated.

**Comment by anandwana001** [comment]
@ShridharGoel Suggestion for further contribution:
Always use the branch name more descriptive. Check our wiki guide here - https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change

**Comment by ShridharGoel** [comment]
@anandwana001 Can you tell which other screenshots need to be added?

**Comment by anandwana001** [comment]
@ShridharGoel the current screenshots in the pr description are of robolectric, you need to add the same for Espresso as well. Need to check all tests are passing in espresso as well.

**Comment by ShridharGoel** [comment]
@anandwana001 Can you tell how to run the tests separately for Espresso? Currently the tests run only for Roboelectric.

**Comment by ShridharGoel** [comment]
@anandwana001 I keep getting this error on running the Espresso tests.

`Test running failed: Instrumentation run failed due to 'Process crashed.'`

**Comment by anandwana001** [comment]
@ShridharGoel 
1. Could you mention which device you are running tests on?
2. Are tests passing when not removing the `testCoroutineDispatchers`?

**Comment by ShridharGoel** [comment]
@anandwana001 

1. I'm using Redmi Note 8.
2. Tests running is failing even in tests in which `testCoroutineDispatchers` hasn't been removed.

**Comment by ShridharGoel** [comment]
Closed by mistake.

**Comment by anandwana001** [comment]
Did you try on an emulator?

**Comment by ShridharGoel** [comment]
No, will try and update.

**Comment by ShridharGoel** [comment]
@anandwana001 Worked fine using emulator. I have updated the screenshots to include screenshots of Espresso tests.

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @ShridharGoel for your first PR at Oppia. 
deferring to @rt4914 for further merge.

**Comment by ShridharGoel** [comment]
@anandwana001 Thank you.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.
